python/demos/demo_christmas_tree.py

```diff
@@ -0,0 +1,219 @@
+# Copyright (C) 2022 Sarah Roggendorf
+#
+# SPDX-License-Identifier:    MIT
+
+import argparse
+import sys
+
+import numpy as np
+from dolfinx import log
+import dolfinx.fem as _fem
+from dolfinx.common import TimingType, list_timings, timing, Timer
+from dolfinx.graph import create_adjacencylist
+from dolfinx.io import XDMFFile
+from mpi4py import MPI
+
+from dolfinx_contact.meshing import (convert_mesh,
+                                     create_christmas_tree_mesh)
+from dolfinx_contact.unbiased.nitsche_unbiased import nitsche_unbiased
+
+if __name__ == "__main__":
+    desc = "Nitsche's method for two elastic bodies using custom assemblers"
+    parser = argparse.ArgumentParser(description=desc,
+                                     formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("--theta", default=1, type=np.float64, dest="theta",
+                        help="Theta parameter for Nitsche, 1 symmetric, -1 skew symmetric, 0 Penalty-like",
+                        choices=[1, -1, 0])
+    parser.add_argument("--gamma", default=10, type=np.float64, dest="gamma",
+                        help="Coercivity/Stabilization parameter for Nitsche condition")
+    parser.add_argument("--quadrature", default=5, type=int, dest="q_degree",
+                        help="Quadrature degree used for contact integrals")
+    _3D = parser.add_mutually_exclusive_group(required=False)
+    _3D.add_argument('--3D', dest='threed', action='store_true',
+                     help="Use 3D mesh", default=False)
+    _timing = parser.add_mutually_exclusive_group(required=False)
+    _timing.add_argument('--timing', dest='timing', action='store_true',
+                         help="List timings", default=False)
+    _ksp = parser.add_mutually_exclusive_group(required=False)
+    _ksp.add_argument('--ksp-view', dest='ksp', action='store_true',
+                      help="List ksp options", default=False)
+    _simplex = parser.add_mutually_exclusive_group(required=False)
+    _simplex.add_argument('--simplex', dest='simplex', action='store_true',
+                          help="Use triangle/tet mesh", default=False)
+    _strain = parser.add_mutually_exclusive_group(required=False)
+    _strain.add_argument('--strain', dest='plane_strain', action='store_true',
+                         help="Use plane strain formulation", default=False)
+    parser.add_argument("--E", default=1e3, type=np.float64, dest="E",
+                        help="Youngs modulus of material")
+    parser.add_argument("--nu", default=0.1, type=np.float64, dest="nu", help="Poisson's ratio")
+    parser.add_argument("--load_steps", default=1, type=np.int32, dest="nload_steps",
+                        help="Number of steps for gradual loading")
+    parser.add_argument("--res", default=0.2, type=np.float64, dest="res",
+                        help="Mesh resolution")
+    parser.add_argument("--outfile", type=str, default=None, required=False,
```

Comment:
Review which of these arguments are actually used.
Of those used you can make some choices on what we want to support for the demo. There is for instance no point of letting the used choose if we use the plane-strain formulation for this demo.